components/post_view/reaction/reaction.jsx

```diff
@@ -116,13 +116,16 @@ export default class Reaction extends React.Component {
 
     static getDerivedStateFromProps(nextProps, prevState) {
         // reaction count has changed, but not by current user
-        if (nextProps.reactionCount !== prevState.displayNumber && nextProps.sortedUsers.currentUserReacted === prevState.userReacted) {
+
+        if (nextProps.reactionCount !== prevState.displayNumber) {
             // set counts relative to current user having reacted
             if (prevState.userReacted) {
                 return {
                     displayNumber: nextProps.reactionCount,
                     reactedNumber: nextProps.reactionCount,
                     unreactedNumber: nextProps.reactionCount - 1,
+                    userReacted: nextProps.sortedUsers.currentUserReacted,
```

Comment:
This doesn't need to be changed now, but I still don't like that these are stored in state instead of just being derived from props when rendered.

Reply:
Same thoughts here. We can open a follow up ticket coz there will be some thinking.